pkg/util/k8s_lookup.go

```diff
@@ -0,0 +1,112 @@
+package util
+
+import (
+	"context"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/interfaces"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+)
+
+//go:generate mockgen -destination=k8s_lookup_mocks.go -package util -source k8s_lookup.go
+
+const MaxChecksWaitingForSpinnakerStability = 1
+
+type Ik8sLookup interface {
+	GetSpinnakerDeployments(instance interfaces.SpinnakerService) ([]appsv1.Deployment, error)
+	GetSpinnakerServiceImageFromDeployment(p v1.PodSpec) string
+	GetPodsByDeployment(instance interfaces.SpinnakerService, deployment appsv1.Deployment) ([]v1.Pod, error)
+	GetReplicaSetByPod(instance interfaces.SpinnakerService, pod v1.Pod) (*appsv1.ReplicaSet, error)
+	HasExceededMaxWaitingTime(instance interfaces.SpinnakerService, pod v1.Pod) (bool, error)
+}
+
+func NewK8sLookup(client client.Client) K8sLookup {
+	return K8sLookup{
+		client: client,
+	}
+}
+
+type K8sLookup struct {
+	client client.Client
+}
+
+// getSpinnakerServices returns the name of the image
+func (l K8sLookup) GetSpinnakerDeployments(instance interfaces.SpinnakerService) ([]appsv1.Deployment, error) {
+	// Get current deployment owned by the service
+	list := &appsv1.DeploymentList{}
+	err := l.client.List(context.TODO(), list, client.InNamespace(instance.GetNamespace()), client.MatchingLabels{"app.kubernetes.io/managed-by": "spinnaker-operator"})
+	if err != nil {
+		return nil, err
+	}
+	if len(list.Items) == 0 {
+		return []appsv1.Deployment{}, nil
+	} else {
+		return list.Items, nil
+	}
+}
+
+// getSpinnakerServiceImageFromDeployment returns the name of the image
+func (l K8sLookup) GetSpinnakerServiceImageFromDeployment(p v1.PodSpec) string {
+	for _, c := range p.Containers {
+		if strings.HasPrefix(c.Name, "spin-") {
```

Comment:
Under what circumstances would the container name start with `spin-`? The few clusters I've been looking at today all have the name of the deployment (e.g. `deploy/spin-front50` has a container named `front50`)

Reply:
To be honest, I don’t know but think we should keep it as is while we get the reason for this check.

Reply:
np, we'll keep an eye on it!